vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder_test.go

```diff
@@ -226,6 +310,10 @@ func TestLoadPods(t *testing.T) {
 				VpaName:   vpa.Name,
 			}
 
+			if tc.expectVpaNotPresent {
+				assert.NotContains(t, clusterState.Vpas, vpaID)
+				return
```

Comment:
@jbartosik I don't understand why the test fails here...
```
    cluster_feeder_test.go:315: missing call(s) to *mocktarget.MockVpaTargetSelectorFetcher.Fetch(is equal to &{{ } {testVpa  testNamespace    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] []  []} {&CrossVersionObjectReference{Kind:dodokind,Name:dotaro,APIVersion:stardust,} <nil> 0xc000111098 []} {0xc0001110b0 []}}) /home/mbertschy/github.com/matthyx/autoscaler/vertical-pod-autoscaler/pkg/target/mock/fetcher_mock.go:57
    cluster_feeder_test.go:315: aborting test due to missing call(s)
```

Reply:
This is because all test cases here set up expectation that they'll Fetch VPA in [line 305](**https://github.com/kubernetes/autoscaler/pull/4799/files#diff-65acf6f86a7495618a847b01cbb7d4f26601545f611fd6ef5702326aeec03341R305):

```
targetSelectorFetcher.EXPECT().Fetch(vpa).Return(tc.selector, tc.fetchSelectorError)
```

and some test cases you add don't fetch VPA object (because they recognize this recommenders issn't supposed to handle the VPA object).

I think the easiest way to handle this is to add a bool to tell each test case if it should set up the expectation, like 

https://github.com/jbartosik/autoscaler/commit/024e1ce8a1fd392ddbb2ab7a4b9d446f2b7cf4f2